macOS/DuckDuckGo/Tab/Model/Tab.swift

```diff
@@ -1326,16 +1326,30 @@ extension Tab/*: NavigationResponder*/ { // to be moved to Tab+Navigation.swift
         permissions.tabDidStartNavigation()
         userInteractionDialog = nil
 
+        let shouldFireNavigationPixel: Bool = {
```

Comment:
since the code grows larger maybe it worths bringing it out to a small NavigationResponder struct and add it to Tab+Navigation (no protocols and dedicated vars, but just nav responder)?